addons/pos_sale/models/pos_order.py

```diff
@@ -55,6 +55,14 @@ def sync_from_ui(self, orders):
             for line in order.lines.filtered(lambda l: l.product_id == order.config_id.down_payment_product_id and l.qty != 0 and (l.sale_order_origin_id or l.refunded_orderline_id.sale_order_origin_id)):
                 sale_lines = line.sale_order_origin_id.order_line or line.refunded_orderline_id.sale_order_origin_id.order_line
                 sale_order_origin = line.sale_order_origin_id or line.refunded_orderline_id.sale_order_origin_id
+                if not any(line.display_type and line.is_downpayment for line in sale_lines):
+                    self.env['sale.order.line'].create(
+                        self.env['sale.advance.payment.inv']._prepare_down_payment_section_values(sale_order_origin)
+                    )
+                # getting last record because SOL is not created yet
+                # Because in multiple down payment with same sale.order we don't know for which pos.order.line SOL is created
+                order_reference = line.sale_order_origin_id.pos_order_line_ids[-1:].name
```

Comment:
```suggestion
                order_reference = line.name
```
I think you can just use the pos.order.line's order reference directly, no need to look for it in the pos.order.line's that are linked to the sale order.

Reply:
Yes that's work..
I have updated the code.

Thank you..